libevmasm/ConstantOptimiser.cpp

```diff
@@ -84,7 +84,10 @@ bigint ConstantOptimisationMethod::simpleRunGas(AssemblyItems const& _items, lan
 	bigint gas = 0;
 	for (AssemblyItem const& item: _items)
 		if (item.type() == Push)
-			gas += GasMeter::runGas(Instruction::PUSH1, _evmVersion);
+		{
+			auto const instruction = (item.data() == u256(0) && _evmVersion.hasPush0()) ? Instruction::PUSH0 : Instruction::PUSH1;
+			gas += GasMeter::runGas(instruction, _evmVersion);
```

Comment:
This is correct - I think we now have this complication in several places, though...
Maybe we should actually add a ``GasMeter::pushGas(item.data(), _evmversion);`` to handle that.

It's a bit funny that ``runGas`` takes the EVM Version anyways, while the comment documenting it states that it calculates ``gas costs for simple instructions with constant gas costs (that do not change with EVM versions)``, but that's already weird on ``develop`` :-).